integration-tests/bats/diff.bats

```diff
@@ -449,6 +449,106 @@ EOF
     [[ ! "$output" =~ "+ | 0" ]] || false
 }
 
+@test "diff: new foreign key added and resolved" {
+    dolt sql <<SQL
+create table parent (i int primary key);
+create table child (j int primary key, foreign key (j) references parent (i));
+SQL
+
+    run dolt diff
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ 'resolved foreign key' ]] || false
+}
+
+@test "diff: new foreign key added and not resolved" {
+    dolt sql <<SQL
+set foreign_key_checks=0;
+create table parent (i int primary key);
+create table child (j int primary key, foreign key (j) references parent (i));
+SQL
+    dolt add -A
+    dolt commit -m "init commit"
+
+    run dolt diff
+    [ "$status" -eq 0 ]
+    [[ ! "$output" =~ 'resolved foreign key' ]] || false
+}
+
+@test "diff: existing foreign key that was resolved is deleted" {
+    dolt sql <<SQL
+create table parent (i int primary key);
+create table child (j int primary key, constraint fk foreign key (j) references parent (i));
+SQL
+    run dolt diff
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ 'resolved foreign key' ]] || false
+
+    dolt add -A
+    dolt commit -m "init commit"
+    dolt sql -q "alter table child drop foreign key fk"
+
+    run dolt diff
+    [ "$status" -eq 0 ]
+    [[ ! "$output" =~ 'resolved foreign key' ]] || false
+}
+
+@test "diff: existing foreign key that was not resolved is deleted" {
+    dolt sql <<SQL
+set foreign_key_checks=0;
+create table parent (i int primary key);
+create table child (j int primary key, constraint fk foreign key (j) references parent (i));
+SQL
+    run dolt diff
+    [ "$status" -eq 0 ]
+    [[ ! "$output" =~ 'resolved foreign key' ]] || false
+
+    dolt add -A
+    dolt commit -m "init commit"
+    dolt sql -q "alter table child drop foreign key fk"
+
+    run dolt diff
+    [ "$status" -eq 0 ]
+    [[ ! "$output" =~ 'resolved foreign key' ]] || false
+}
+
+@test "diff: existing foreign key that was resolved is modified" {
+    dolt sql <<SQL
+create table parent (i int primary key);
+create table child (j int primary key, constraint fk foreign key (j) references parent (i));
+SQL
+    run dolt diff
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ 'resolved foreign key' ]] || false
+
+    dolt add -A
+    dolt commit -m "init commit"
+    dolt sql -q "alter table child drop foreign key fk"
+    dolt sql -q "alter table child add constraint fk foreign key (j) references parent (i)"
```

Comment:
This actually doesn't register as a diff at all, so it doesn't show `resolved foreign key` because there was no diff in the first place. You need to modify the calling column or something to actually create a change in the foreign key.